csharp/src/Ice/MultiStreamSocket.cs

```diff
@@ -32,6 +32,7 @@ public abstract class MultiStreamSocket : IDisposable
         public bool IsIncoming { get; }
 
         internal int IncomingFrameMaxSize { get; }
+        internal int? OutgoingFrameMaxSize { get; set; }
```

Comment:
Would be good to add a doc-comment describing these properties.

My understanding in the OutgoingFrameMaxSize is the peer's IncomingFrameMaxSize and therefore is only set during connection establishment.

Reply:
We could also rename OutgoingFrameMaxSize => PeerIncomingFrameMaxSize?